cmd/eval_test.go

```diff
@@ -1312,6 +1312,116 @@ time.clock(input.y, time.clock(input.x))
 	}
 }
 
+func TestEvalDiscardOutput(t *testing.T) {
```

Comment:
It would be helpful if we add a test case for the scenario when the result is `nil`.

Reply:
I have added that as a second case in that tests struct

Reply:
I would have expected `opa eval '1*2+3' -format=discard` to produce nothing unless something like `--profile` was set. @charlieegan3 and @26tanishabanik am I missing something here?

Reply:
I think, if I am remember properly, the idea was to set it empty if the result is itself nil

Reply:
I see three desirable behaviours as goals for this:

1. Undefined results are not shown as discarded as they were never there in the first place.
2. Discarded results appear differently from undefined results to make the distinction clear.
3. `--format=discard` will always produce valid JSON. From the issue (https://github.com/open-policy-agent/opa/issues/5863) it's meant to be the 'opposite' of `--format=raw` which removes all structure around the result.

In the `go run main.go eval '1*2+3' --format=discard` example, the result defined and so is discarded in the output. We can't return `{}` since it'd break goal 2.

```
$ go run main.go eval '1*2+3' --format=discard 
{
  "result": "discarded"
}
```

I think this handling of undefined results is also correct:

```
$ go run main.go eval '1' --format=discard 
{
  "result": "discarded"
}
$ go run main.go eval 'input.undefined' --format=discard 
{}
```

Reply:
How should we proceed further?

Reply:
I'll let @ashutosh-narkar weigh in to check we're all in agreement on the desired outputs first. I can work with you again on this tomorrow if we need to make any updates 🙂

Reply:
Sure, thanks 🙂

Reply:
This sounds fine. Thanks for the clarification @26tanishabanik and @charlieegan3.